scripts/WFCORE-5261-common-conf.adoc

```diff
@@ -0,0 +1,104 @@
+= Add Common Script Configuration File
+:author:            James R. Perkins
+:email:             jperkins@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+There is a desire to have a common script configuration file, `common.conf`. This will be called before the standalone
+or domain script configuration file. The other scripts will also invoke the common script configuration file. It allows
+things like common environment variables to be set by users regardless of the script being used.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.redhat.com/browse/WFCORE-5261[WFCORE-5261]
+* https://issues.redhat.com/browse/WFLY-14399[WFLY-14399]
+
+=== Related Issues
+
+* https://issues.redhat.com/browse/EAP7-1251[EAP7-1251]
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+* TBD
+
+=== Testing By
+
+* [ ] Engineering
+
+* [ ] QE
+
+=== Affected Projects or Components
+
+* WildFly Core
+* WildFly
+* Any project or component which invokes script files
+
+=== Other Interested Projects
+
+=== Relevant Installation Types
+* [x] Traditional standalone server (unzipped or provisioned by Galleon)
+
+* [x] Managed domain
+
+* [x] OpenShift s2i
+
+* [ ] Bootable jar
+
+== Requirements
+
+=== Hard Requirements
+
+* The common script should be invoked before the standard script configuration files.
+* This will need to work on all supported platforms.
+** `common.conf`
+** `common.conf.bat`
+** `common.conf.ps1`
+* The environment variable `COMMON_CONF` can be used to override the path to the file.
+** By default, `$JBOSS_HOME/bin/common.conf` will be used.
+* All scripts in the `$JBOSS_HOME/bin` directory will invoke this script configuration file.
+** Note this includes scripts in WildFly Core and WildFly.
+
+=== Nice-to-Have Requirements
+
+=== Non-Requirements
+
+* The `common.conf`, `common.conf.bat` and `common.conf.ps1` files do not need to be present in default distribution.
```

Comment:
In case COMMON_CONF is defined but the referenced file does not exist or is not accessible, will there be a warning informing a user about such suspicious situation?

Reply:
I did not implement it that way, but that is a good idea.

Reply:
@jamezp: Do we have this implemented now? If yes, what error message does the system provide?